src/unix/cygwin/mod.rs

```diff
@@ -0,0 +1,2449 @@
+        ((*set).fds_bits[fd / size] & (1 << (fd % size))) != 0
+    }
+
+    pub fn FD_SET(fd: c_int, set: *mut fd_set) {
+        let fd = fd as usize;
+        let size = core::mem::size_of_val(&(*set).fds_bits[0]) * 8;
+        (*set).fds_bits[fd / size] |= 1 << (fd % size);
+    }
+
+    pub fn FD_ZERO(set: *mut fd_set) {
+        for slot in (*set).fds_bits.iter_mut() {
+            *slot = 0;
+        }
+    }
+
+    pub fn CPU_ALLOC_SIZE(count: c_int) -> size_t {
+        let _dummy: cpu_set_t = cpu_set_t { bits: [0; 16] };
+        let size_in_bits = 8 * core::mem::size_of_val(&_dummy.bits[0]);
+        ((count as size_t + size_in_bits - 1) / 8) as size_t
+    }
+
+    pub fn CPU_COUNT_S(size: usize, cpuset: &cpu_set_t) -> c_int {
+        let mut s: u32 = 0;
+        let size_of_mask = core::mem::size_of_val(&cpuset.bits[0]);
+        for i in cpuset.bits[..(size / size_of_mask)].iter() {
+            s += i.count_ones();
+        }
+        s as c_int
+    }
+
+    pub fn CPU_ZERO(cpuset: &mut cpu_set_t) {
+        for slot in cpuset.bits.iter_mut() {
+            *slot = 0;
+        }
+    }
+    pub fn CPU_SET(cpu: usize, cpuset: &mut cpu_set_t) {
+        let size_in_bits = 8 * core::mem::size_of_val(&cpuset.bits[0]);
+        if cpu < size_in_bits {
+            let (idx, offset) = (cpu / size_in_bits, cpu % size_in_bits);
+            cpuset.bits[idx] |= 1 << offset;
+        }
+    }
+
+    pub fn CPU_CLR(cpu: usize, cpuset: &mut cpu_set_t) {
+        let size_in_bits = 8 * core::mem::size_of_val(&cpuset.bits[0]);
+        if cpu < size_in_bits {
+            let (idx, offset) = (cpu / size_in_bits, cpu % size_in_bits);
+            cpuset.bits[idx] &= !(1 << offset);
+        }
+    }
+
+    pub fn CPU_ISSET(cpu: usize, cpuset: &cpu_set_t) -> bool {
+        let size_in_bits = 8 * core::mem::size_of_val(&cpuset.bits[0]);
+        if cpu < size_in_bits {
+            let (idx, offset) = (cpu / size_in_bits, cpu % size_in_bits);
+            0 != (cpuset.bits[idx] & (1 << offset))
+        } else {
+            false
+        }
+    }
+
+    pub fn CPU_COUNT(cpuset: &cpu_set_t) -> c_int {
+        CPU_COUNT_S(::core::mem::size_of::<cpu_set_t>(), cpuset)
+    }
+
+    pub fn CPU_EQUAL(set1: &cpu_set_t, set2: &cpu_set_t) -> bool {
+        set1.bits == set2.bits
+    }
+
+    pub fn major(dev: dev_t) -> c_uint {
+        ((dev >> 16) & 0xffff) as c_uint
+    }
+
+    pub fn minor(dev: dev_t) -> c_uint {
+        (dev & 0xffff) as c_uint
+    }
+
+    pub fn CMSG_LEN(length: c_uint) -> c_uint {
+        CMSG_ALIGN(::core::mem::size_of::<cmsghdr>()) as c_uint + length
+    }
+
+    pub {const} fn CMSG_SPACE(length: c_uint) -> c_uint {
+        (CMSG_ALIGN(length as usize) + CMSG_ALIGN(::core::mem::size_of::<cmsghdr>())) as c_uint
+    }
+
+    pub fn CMSG_FIRSTHDR(mhdr: *const msghdr) -> *mut cmsghdr {
+        if (*mhdr).msg_controllen as usize >= core::mem::size_of::<cmsghdr>() {
+            (*mhdr).msg_control as *mut cmsghdr
```

Comment:
Prefer `.cast::<cmsghdr>()` over `as` casting with `.cast_mut()` if needed in order to avoid silently changing mutability. Applies a few places.